File: database/src/main/java/com/coopra/database/Converters.java
package com.coopra.database;

import com.coopra.database.entities.User;
import com.google.gson.Gson;

import androidx.room.TypeConverter;

public class Converters {
    @TypeConverter
    public String fromUser(User user) {
        Gson gson = new Gson();
        String json = gson.toJson(user);
        return json;
    }

    @TypeConverter
    public User toUser(String userJson) {
        Gson gson = new Gson();
        User user = gson.fromJson(userJson, User.class);
        return user;
    }
}

File: database/src/main/java/com/coopra/database/Converters.kt
package com.coopra.database

import androidx.room.TypeConverter
import com.coopra.database.entities.User
import com.google.gson.Gson

class Converters {
    @TypeConverter
    fun fromUser(user: User): String {
        val gson = Gson()
        return gson.toJson(user)
    }

    @TypeConverter
    fun toUser(userJson: String): User {
        val gson = Gson()
        return gson.fromJson(userJson, User::class.java)
    }

    @TypeConverter
    fun fromWaveform(waveform: IntArray): String {
        val gson = Gson()
        return gson.toJson(waveform)
    }

    @TypeConverter
    fun toWaveform(waveformJson: String): IntArray {
        val gson = Gson()
        return gson.fromJson(waveformJson, IntArray::class.java)
    }
}